server/controllers/userController.js

```diff
@@ -0,0 +1,64 @@
+const db = require("../models");
+
+// Defining methods for the userController
+module.exports = {
+  getUser: (req, res, next) => {
+    console.log("we are in getuser")
+    // console.log(req.user);
+    if (req.user) {
+      return res.json({ user: req.user });
+    } else {
+      return res.json({ user: null });
+    }
+  },
+  register: (req, res) => {
+    const { firstName, lastName, username, password } = req.body;
+    console.log("we are in register")
+    db.User.findOne({ 'username': username }, (err, userMatch) => {
+      if (userMatch) {
+        return res.json({
+          error: `Sorry, already a user with the username: ${username}`
+        });
+      }
+      const newUser = new db.User({
+        'firstName': firstName,
+        'lastName': lastName,
+        'username': username,
+        'password': password,
```

Comment:
I think for the purposes of this project, this makes sense, but you'll want to make this more defensive against attackers that may want to pwn your DB.